### DISCUSSION TITLE
Selective Channel Export Confusion

### DISCUSSION THREAD

**Discussion by DamianHarty** [danielhrisca/asammdf#847] [Q&A]
O learned denizens, I humbly prostrate myself before your collective wisdom.

I am really new to Python, GitHub and all sorts of things, so please forgive me if I am gauche.

I downloaded asammdf and have been using it to immediate and vivid effect. I can browse files, make a configuration I want, force axis scaling, use subplots, etc. It's completely marvellous for reviewing logged data, which for me is characterised by about a dozen channels I care about among about a thousand I don't, across multiple CAN buses and without access to commercial software tools. I can skip and frolic and make screenshots and everything I want. It's an intuitive delight.

Windows 10, Python 3.11.2, asammdf 7.3.5.

Now, the baffled bit. The first file I opened, I decided to export some of the channels. I selected them and went to the modify & export tab. I resampled at 100 Hz (plenty good enough for what I'm doing), set a single time base, etc. I got a CSV file just as I wanted it - the signal labels, the delimiter, the signals of interest, etc, etc.

Then I went to another file, and try as I might I couldn't get it to export only the signals I am interested in. I saved a channel selection list from the first file and loaded it with the second file. When I set the drop down to "Selected Channels Only" it resolutely exports every one of the ~1000 channels in the file. 

In pasting this query I set out to reproduce the behaviour with the original file only to find it, too would only export all channels. 

Then I changed the drop-down to "Natural Sort" and *now* it only exports the selected channels, as anticipated.

Is this how it should work, is there some underlying logic I am missing, or is it just a lovable quirk in a free and fabulously functional piece of software? 

Thanks in advance!

Damian

**Comment by danielhrisca** [comment]
The selection that you are making in the `Channels` tab has nothing to do with the one used by the `Modify & Export` tab

**Reply by DamianHarty** [reply]
Sorry that my post was ambiguous. The behaviour I describe in the later part of the post was in the Modify & Export tab. I saved a .lab file and applied it to the new file, and it selected the channels of interest successfully. 

The confusion is that *in the Modify & Export tab*, when Selected Channels Only is chosen as the sort order, and only the 12 channels are displayed in the panel in the middle of the screen, all channels export.

When Natural Sort Order is chosen without changing any selections then the behaviour is as expected.

**Reply by danielhrisca** [reply]
Please try out version 7.3.11 from PyPI

**Reply by DamianHarty** [reply]
Alas, the behaviour is identical with 7.3.11. It's not a drama, because the workaround is easy.

**Reply by danielhrisca** [reply]
what about 7.3.12 ? 😅

**Reply by DamianHarty** [reply]
7.3.12 now works as I expect, whatever I select as the sort order. Huzzah!
